src/hooks/use-short-address.tsx

```diff
@@ -0,0 +1,35 @@
+import { useState, useMemo } from "react";
+import { useNotifications } from "./use-notifications";
+import { NotificationStatusType } from "@/comps/Notifications";
+
+export function useShortAddress(address: string) {
+  const [isHovered, setIsHovered] = useState(false);
+  const { notify } = useNotifications();
+
+  const onMouseEnter = () => setIsHovered(true);
+  const onMouseLeave = () => setIsHovered(false);
+  const copyOnClick = () => {
+    navigator.clipboard.writeText(address);
+    notify({
+      message: "Address copied to clipboard",
+      type: NotificationStatusType.SUCCESS,
+    });
+  };
+
+  const shortenedAddress = useMemo(() => {
+    if (address.length > 10) {
```

Comment:
nit: it shouldn't matter given the nature of the addresses, but still
```suggestion
    if (address.length > 20) {
```